tool/tsh/common/tsh_test.go

```diff
@@ -1772,227 +1772,256 @@ func (o *output) String() string {
+			// wait for auth to see nodes
+			require.Eventually(t, hasNodes(sshHostID, sshHostID2, sshHostIDNoAccess),
+				10*time.Second, 100*time.Millisecond, "nodes never showed up")
 
-	err = Run(ctx, []string{
-		"login",
-		"--insecure",
-		"--proxy", proxyAddr.String(),
-		"--user", "alice",
-	}, setHomePath(tmpHomePath), setMockSSOLogin(rootAuth.GetAuthServer(), alice, connector.GetName()))
-	require.NoError(t, err)
-
-	// won't request if can't list node
-	err = Run(ctx, []string{
-		"ssh",
-		"--insecure",
-		"--request-reason", "reason here to bypass prompt",
-		fmt.Sprintf("%s@%s", user.Username, sshHostnameNoAccess),
-		"echo", "test",
-	}, setHomePath(tmpHomePath))
-	require.Error(t, err)
+			err = Run(ctx, []string{
+				"login",
+				"--insecure",
+				"--proxy", proxyAddr.String(),
+				"--user", "alice",
+			}, setHomePath(tmpHomePath), setMockSSOLogin(rootAuth.GetAuthServer(), alice, connector.GetName()))
+			require.NoError(t, err)
 
-	// won't request if can't login with username
-	err = Run(ctx, []string{
-		"ssh",
-		"--insecure",
-		"--request-reason", "reason here to bypass prompt",
-		fmt.Sprintf("%s@%s", "not-a-username", sshHostname),
-		"echo", "test",
-	}, setHomePath(tmpHomePath))
-	require.Error(t, err)
+			// won't request if can't list node
+			err = Run(ctx, []string{
+				"ssh",
+				"--insecure",
+				"--request-mode", tc.requestMode,
+				"--request-reason", "reason here to bypass prompt",
+				fmt.Sprintf("%s@%s", user.Username, sshHostnameNoAccess),
+				"echo", "test",
+			}, setHomePath(tmpHomePath))
+			require.Error(t, err)
 
-	// won't request to non-existent node
-	err = Run(ctx, []string{
-		"ssh",
-		"--insecure",
-		fmt.Sprintf("%s@unknown", user.Username),
-		"echo", "test",
-	}, setHomePath(tmpHomePath))
-	require.Error(t, err)
+			// won't request if can't login with username
+			err = Run(ctx, []string{
+				"ssh",
+				"--insecure",
+				"--request-mode", tc.requestMode,
+				"--request-reason", "reason here to bypass prompt",
+				fmt.Sprintf("%s@%s", "not-a-username", sshHostname),
+				"echo", "test",
+			}, setHomePath(tmpHomePath))
+			require.Error(t, err)
 
-	// approve all requests as they're created
-	errChan := make(chan error)
-	t.Cleanup(func() {
-		require.ErrorIs(t, <-errChan, context.Canceled, "unexpected error from approveAllAccessRequests")
-	})
-	go func() {
-		err := approveAllAccessRequests(ctx, rootAuth.GetAuthServer())
-		// Cancel the context, so Run calls don't block
-		cancel()
-		errChan <- err
-	}()
+			// won't request to non-existent node
+			err = Run(ctx, []string{
+				"ssh",
+				"--insecure",
+				"--request-mode", tc.requestMode,
+				fmt.Sprintf("%s@unknown", user.Username),
+				"echo", "test",
+			}, setHomePath(tmpHomePath))
+			require.Error(t, err)
 
-	// won't request if explicitly disabled
-	err = Run(ctx, []string{
-		"ssh",
-		"--insecure",
-		"--request-reason", "reason here to bypass prompt",
-		"--disable-access-request",
```

Comment:
can we keep this test case since it's still supported?

Reply:
It's still there, the diff is just making it hard to see.

Reply:
you have a testcase with `--request-mode=off`, I don't see one with `--disable-access-request`. Since that flag is still supported I think we should keep the test (keep both)